Packs/Rapid7_Nexpose/Integrations/Rapid7_Nexpose/Rapid7_Nexpose.py

```diff
@@ -6458,6 +6695,907 @@ def get_list_asset_group_command(
+
+
+def update_state_checkpoint(collector_type: str, changes: Dict[str, Any]) -> None:
+    """
+    Retrieves the current integration context, updates the specified collector's
+    state with the provided changes, and saves the updated context.
+
+    Args:
+        collector_type: The top-level key to update ('asset' or 'vulnerability').
+        changes: A dictionary of {key: value} updates to apply.
+    """
+
+    # 1. Get the current context
+    current_context = get_integration_context()
+
+    # 2. Apply the update using the helper function
+    _apply_collector_changes(current_context, collector_type, changes)
+
+    # 3. Set the updated context back to the platform
+    set_integration_context(current_context)
+    demisto.debug(f"State checkpoint saved for '{collector_type}'. New line: {current_context[collector_type]}")  # noqa: E501
+
+
+#################################################
+# copy from akamai
+#################################################
+
+
+async def process_and_send_events_to_xsiam(
+    events: list[str],
+    total_lines_after_batch_raw: int | None,
+    batch_counter: int,
+    new_total_records: int,
+    event_type: str,
+    snapshot_id: str,
+    items_count: int = 1,
+):
+    """Send the events to CSP's send_events_to_xsiam.
+    save the offset after the execution is done and update the module health.
+
+    Args:
+        events (list[str]): The list of json serialized events.
+        should_skip_decode_events (bool): Whether we should skip serializing and decoding events or not.
+        offset (str | None): The offset hash.
+        counter (int): The current execution number.
+    """
+    demisto.debug(f"Running in interval = {batch_counter}. got {len(events)} events, moving to processing events data.")  # noqa: E501
+    processed_events = events
+
+    size_in_bytes = sum(len(line.encode("utf-8")) for line in events)
+
+    tasks = send_events_to_xsiam_rapid7(
+        processed_events,
+        VENDOR.get(event_type),
+        PRODUCT.get(event_type),
+        should_update_health_module=False,
+        chunk_size=size_in_bytes,
+        send_events_asynchronously=True,
+        url_key="host",
+        data_format="json",
+        data_size_expected_to_split_evenly=True,
+        counter=batch_counter,
+        items_count=items_count,
+        snapshot_id=snapshot_id,
+    )
+    demisto.debug(f"Running in interval = {batch_counter}. Finished executing send_events_to_xsiam, waiting for tasks to end.")
+    await asyncio.gather(*tasks)
+    demisto.debug(f"Running in interval = {batch_counter}. Finished gathering all tasks.")
+    demisto.debug(f"Running in interval = {batch_counter}. Updating module health.")
+
+    update_state_checkpoint(
+        event_type,
+        {"last_sent_line": total_lines_after_batch_raw, "total_records_ingested": new_total_records, "snapshot_id": snapshot_id},
+    )
+    demisto.updateModuleHealth({"assetsPulled": len(processed_events)})
```

Comment:
Don't think this is necessary at this part.The 
snapshot will be updated only once the whole flow is done.

Reply:
Will we have any indication on the UI part on the number of fetched assets?

Reply:
Removing as there is no indication whatsoever